tinygrad/runtime/ops_cpu.py

```diff
@@ -45,6 +44,8 @@ def mulacc(a, b, new_shape):
 class RawNumpyBuffer(RawBuffer):
   def __init__(self, size:int, dtype:DType, buf:Optional[np.ndarray]=None): super().__init__(size, dtype, buf if buf is not None else np.empty([size], dtype.np))
   @classmethod
-  def fromCPU(cls, x): return cls(x.size, dtypes.from_np(x.dtype), x)
+  def fromCPU(cls, x): 
```

Comment:
This should not be here. I think the promote types might be a bug too. Can you submit this is as a separate PR?